src/mtype.h

```diff
@@ -208,12 +216,13 @@ struct mtype {
         std::string death_drops;
         float luminance;           // 0 is default, >0 gives luminance to lightmap
         int hp;
-        std::vector<unsigned int> sp_freq;     // How long sp_attack takes to charge
+        // special attack frequencies and function pointers
+        std::unordered_map<std::string, mtype_special_attack> special_attacks;
+        std::vector<std::string> special_attacks_names; // names of attacks, in json load order
```

Comment:
I did consider putting the load order into a member of the mtype_special_attack struct. However, that would require a search through the list of special attacks when trying to do things in order during old save loading. I chose this approach, a secondary vector, so that the loading could simply index into the vector.